### PR TITLE
Remove get_device

### PR DESCRIPTION
**Pull request by mitmul**
get_device has been marked as deprecated, so we can remove it in v2. It takes over this PR: #2243

### CODE REVIEW

File: chainer/cuda.py [hunk 1]
```diff
@@ -337,10 +301,10 @@ def copy(array, out=None, out_device=None, stream=None):
     if out is None:
         if out_device is None:
             out_device = array
-        with get_device(out_device):
+        with get_device_from_id(out_device):
```
Comment (niboshi) [inline chainer/cuda.py]: Should this be `_get_device`? I think `out_device` can be either array or id.

**Comment by mitmul** [comment]
@niboshi Thank you very much for your quick review. I fixed them.

**Comment by niboshi** [comment]
Thank you. You also need to fix 2 more `get_device()` lines in `optimizer.py`.

**Comment by niboshi** [comment]
@unnonouno Can you check and merge?

**Comment by beam2d** [comment]
We decided to postpone this change. 1) We need a longer period to remove the deprecated `get_device`, so the removal will be done in the next major. 2) Removing the use of `get_device` from Chainer's internal code can be done in the revision update, so we will make it in 2.0.1 or later.

**Comment by mitmul** [comment]
It's been replaced with #2890

**Pull request closed without merging**